common/lib/monitoring.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-console */
```

Comment:
Is this monitoring part of the PR, or is it a holdover from your previous promtheus PR?

Reply:
> Beyond that, the last 2 commits show possible automated/unified handling of logging+metrics for the services

Sorry, the monitoring part is just a demonstration of how we could add it in for all methods we wanted as essentially config. (What with the numerous rather gnarly rebases I've had to do, I didn't fancy spinning off another branch on top of this one)

I would actually remove those commits from this PR and put them in a new one.